src/modules/user/user.service.ts

```diff
@@ -71,10 +106,10 @@ export class UserService {
 			};
 		} catch (error) {
 			if (error?.response) throw error;
-			throwHttpException(RESPONSE_TYPES.SERVER_ERROR, "Failed to create user");
+			throwHttpException(RESPONSE_TYPES.SERVER_ERROR, `Failed to create user`);
```

Comment:
**Update on my brainfart:** catch is enough only in the top-level method. No need to have a catch in handleGetUserByEmail() as the error travels all the way top to the catch block. I would prefer to have custom exceptions like
 
```
class InvalidEmailError extends Error {
    constructor(message: string) {
        super(message);
        this.name = "InvalidEmailError";
    }
}

class UserCreationFailedError extends Error {
    constructor(message: string) {
        super(message);
        this.name = "UserCreationFailedError";
    }
}
```

This way we can have a nice error handling like this:

```
} catch (error) {
            if (error instanceof InvalidEmailError) {
                throw new UserCreationFailedError("User creation failed due to invalid email");
            } else {
                throw new UserCreationFailedError("User creation failed");
            }
        }
```